routes/pointofinterest/route.js

```diff
@@ -1,22 +1,73 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable prefer-const */
+/* eslint-disable func-names */
 const express = require('express');
 
 const router = express.Router();
+const bodyParser = require('body-parser');
+
 const connection = require('../../conf');
 const transformPoiSampleJson = require('../../functions/transformPoiSampleJson');
-const transformPoiIdJson = require('../../functions/transformPoiIdJson');
-const getPoiInfosById = require('../../queries/getPoiInfosById');
 const getSamplePoisInfos = require('../../queries/getSamplePoisInfos');
-const createNewPoi = require('../../queries/createNewPoi');
+const getSamplePoisInfosCriteria = require('../../queries/getSamplePoisInfosCriteria');
+const { createNewPoi } = require('../../queries/createNewPoi');
 const getKeywords = require('../../queries/getKeywords');
 const getFilteredPoi = require('../../queries/getFilteredPoi');
 
+
+// Support JSON-encoded bodies
+router.use(bodyParser.json());
+
+// Support URL-encoded bodies
+router.use(bodyParser.urlencoded({
+  extended: true,
+}));
+
+// allow to get and post content from localhost to localhost
 router.use((req, res, next) => {
   res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
   next();
 });
 
-router.get('/sample', (req, res) => {
-  connection.query(getSamplePoisInfos, (err, datas) => {
+router.use((req, res, next) => {
+  let { send } = res;
+  let sent = false;
+  res.send = function (data) {
+    if (sent) return;
+    send.bind(res)(data);
+    sent = true;
+  };
+  next();
+});
+
+router.post('/', (req, res) => {
+  const formData = req.body;
+  connection.query(createNewPoi, [formData], (err) => {
+    if (err) {
+      res.status(500).send(`Erreur lors de la récupération des données : ${err}`);
+    } else {
+      res.sendStatus(200);
+    }
+  });
+});
+
+// get two closest pois per keyword in a 3 km radius around the user at loading
+router.get('/sample/:latitude/:longitude', (req, res) => {
+  const latToUse = req.params.latitude;
+  const longToUse = req.params.longitude;
+  connection.query(`${getSamplePoisInfos} WHERE keyword.name = 'Sport' ${getSamplePoisInfosCriteria} 
+  UNION (${getSamplePoisInfos} WHERE keyword.name = 'Nature' ${getSamplePoisInfosCriteria})
+  UNION (${getSamplePoisInfos} WHERE keyword.name = 'Attraction' ${getSamplePoisInfosCriteria})
+  UNION (${getSamplePoisInfos} WHERE keyword.name = 'Monument' ${getSamplePoisInfosCriteria})
+  UNION (${getSamplePoisInfos} WHERE keyword.name = 'Utilitaire' ${getSamplePoisInfosCriteria})
+  UNION (${getSamplePoisInfos} WHERE keyword.name = 'Hygiène' ${getSamplePoisInfosCriteria});`,
```

Comment:
Why the query isn't in a variable like all the other one ?

Reply:
The problem has been corrected, SQL query has been set in a variable.